### DISCUSSION TITLE
Add a quality option for both iOS and Android. 画質を選択できる機能を作りたいです。

### DISCUSSION THREAD

**Discussion by tian-zhihui** [shogo4405/HaishinKit.dart#30] [Ideas]
iOSの場合、デフォルトのsessionPresetはAVCaptureSessionPreset.mediumになっているため、
高い解像度を設定しても画質が低いのままです。
この問題を解決するために、画質を設定するインタフェースを開発したいと考えています。
iOSとAndroidの仕様が異なる部分があり、OSごとに違う設定項目にするか、画質（Quality）という共通項目にして、
AndroidとiOSネイティブコードに画質及び解像度に応じて、それぞれのAPIを呼び出すかは迷っています。
ライブラリーのオーナーとして @shogo4405  shogoさんのご意見いただけますか？

In the case of iOS, the default sessionPreset is AVCaptureSessionPreset.medium, so even when setting a high resolution, the video quality remains low.
To address this issue, I would like to develop an interface to adjust the capture quality.
However, there are differences in specifications between iOS and Android. I am unsure whether to have separate settings for each OS or have a common setting for capture quality, considering both quality and resolution, and call the respective APIs in the Android and iOS native code. 
As the owner of this library, could you please give me some suggestions?

**Comment by shogo4405** [comment]
iOSもAndroidも基本的にカメラの入力解像度はデバイス単位になっています。
iOSは、一括で指定できるように sesionPreset があるようです。

したがって、
1. iOS専用に sessionPreset のみ変更するオプションをつくる
2. カメラ単位に設定可能ように設定する
    1. stream.captureSessing を経由して
        1. iOSだと CameraDeviceのactiveFormatを変更可能にする
        2. Androidだと、Camera2Sourceの resolution を設定可能にする

という選択をとることになると思います。2だと、メインのライブラリ側の修正が必要なため
1のsessionPresetのみを変更できるようにすると良いと思います。
